cli/cppcheckexecutor.cpp

```diff
@@ -55,9 +55,8 @@
 #elif !defined(__OpenBSD__)
 #   include <ucontext.h>
 #endif
-#ifdef __linux__
+#if defined(__linux__) && defined(REG_ERR)
```

Comment:
hm I am not sure about this change. you can remove the `&& defined(REG_ERR)` and it still compiles properly, therefore this change makes the code more cludgy.

Reply:
The code that requires this actually needs to be split. I just wanted to adjust the includes in this and not change any code.

Reply:
I assume you mean this code:

    #if defined(__linux__) && defined(REG_ERR)
        const ucontext_t* const uc = reinterpret_cast<const ucontext_t*>(context);
        killid = (pid_t) syscall(SYS_gettid);
        if (uc) {
            type = (int)uc->uc_mcontext.gregs[REG_ERR] & 2;
        }
    #else
        UNUSED(context);
        killid = getpid();
    #endif

that code is fine imho. you can still remove the `&& defined(REG_ERR)` and compile the code properly.

I think you take include-what-you-use too seriously.

I like the more generic cleanups when a include can be removed completely.

but I rather not make the ifdefs more complex.

Reply:
I will split that code.